File: C & C++/Remove Nth Node From End of List.cpp
#include<stddef.h>
  //Definition for singly-linked list.
  struct ListNode {
      int val;
      ListNode *next;
      ListNode() : val(0), next(nullptr) {}
      ListNode(int x) : val(x), next(nullptr) {}
      ListNode(int x, ListNode *next) : val(x), next(next) {}
  };
 
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* ts = head;
        ListNode* tf = head;
        if (head->next == NULL)
            return NULL;
        for( int i = 1; i <= n+1 ; i++)
           {
            if ((tf->next == NULL) &&  (i == n))
            {       
                head = head->next;
                return head;
            }
            tf = tf->next;
           }    
        while( tf != NULL)
        {
            ts=ts->next;
            tf=tf->next;
        }
        ts->next=ts->next->next;
        return head;
        
    }
};

File: Python/Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        ts = ListNode()
        tf = ListNode()
        ts = head
        tf = head
        for i in range(n+1):
            if tf.next == None and i == (n-1):
                head = head.next
                return head
            tf = tf.next
        
        while( tf != None):
            ts = ts.next
            tf = tf.next
        
        ts.next = ts.next.next
        return head